compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -963,6 +965,19 @@ public void visit(BLangArrayType arrayType) {
 
     @Override
     public void visit(BLangUserDefinedType userDefinedType) {
+        BType type = userDefinedType.getBType();
+        if (type == null) {
+            result = userDefinedType;
+            return;
+        }
+        if (type.tag == TypeTags.RECORD) {
+            ((BRecordType) type).enclMapSymbols = collectClosureMapSymbols(env, false);
+        } else if (type.tag == TypeTags.ERROR) {
+            BType detailType = ((BErrorType) type).detailType;
+            if (detailType.tag == TypeTags.RECORD) {
+                ((BRecordType) detailType).enclMapSymbols = collectClosureMapSymbols(env, false);
+            }
+        }
```

Comment:
What about other user-defined types that may have the record? Don't we get here for them?

Reply:
Any update on this?

Reply:
This logic moves to record type node visitor. https://github.com/ballerina-platform/ballerina-lang/pull/39101/commits/0b954eeccda6e6f7a813c79b9edab058f4c3d159